Sources/StreamChat/Models/PushDevice.swift

```diff
@@ -6,25 +6,46 @@ import Foundation
 
 /// The information required to register a device for push notifications.
 public struct PushDevice {
+    /// Device token used to register a device for push notifications
     public let deviceId: DeviceId
-    public let provider: PushProvider
 
-    private init(deviceId: DeviceId, provider: PushProvider) {
+    /// Provider of the remote push notification service (eg. `APN`, `Firebase`)
+    public let pushProvider: PushProvider
+
+    /// Name of your push configuration in the Stream's dashboard.
+    public let providerName: String?
```

Comment:
Why do we call configuration a provider? That's a bit confusing IMO, because currently when we say `provider` we mean either APN or Firebase

Reply:
This follows the naming adopted by the backend. It's good question, I'll pipe it up to workgroup. At this moment the naming change would cause `iOS` to diverge from the rest of the SDKs.

Reply:
provider is APN, firebase, etc. but they can be multiple whose name defined by users such as staging, prod, etc.